### DISCUSSION TITLE
간단한 NPM 라이브러리 배포해보기 2탄 (CLI 간단한 라이브러리 만들기 & 모든 환경 지원, ESM, CJS, TS) - 정현수 기술 블로그

### DISCUSSION THREAD

**Discussion by giscus** [junghyeonsu/junghyeonsu.dev.comment#62] [Announcements]
# 간단한 NPM 라이브러리 배포해보기 2탄 (CLI 간단한 라이브러리 만들기 & 모든 환경 지원, ESM, CJS, TS) - 정현수 기술 블로그

ESM, CJS, TS를 모두 지원하는 라이브러리를 만들어보고, CLI를 간단하게 만들어보겠습니다.

https://junghyeonsu.com/posts/deploy-simple-npm-library/

**Comment by kh8640** [comment]
글 너무 잘 봤습니다 
라이브러리 배포에 대해 많이 배웠어요
npm에서는 비공개로 배포하려면 사용하는 인원당 7달러의 유료 플랜밖에 없는 것 같은데
비공개로 라이브러리를 배포하실 때는 어떻게 배포하셨는지 여쭤봐도 될까요?

**Reply by junghyeonsu** [reply]
private npm 배포는 말그대로 유료 플랜이에요!
말씀주신대로 인당 7달러이기 때문에 개인적으로 사용하기에는 많이 무리가 있어요.

비공개로 라이브러리를 운영하기보다 최대한 내부적인 도메인 로직을 빼고
모두가 사용할 수 있는 인터페이스를 고민해서 오픈소스화 한 다음에
해당 라이브러리를 내부에서 도메인 지식을 따로 붙여서 사용할 수 있는 방향은 어떠신가요!

**Comment by henlyshin** [comment]
잘보고 갑니다. 좀 다른 질문이 있는데.. 
저런 귀여운 이미지는 AI에서 가져오신건가요? ㅎㅎ 포스트 하실때 썸네일은 어찌 하시는지 궁금합니다.
